packages/shared/test/shared-secret.test.ts

```diff
@@ -0,0 +1,221 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import {
+  DecryptionError,
+  DecryptionErrorCode,
+  EncryptionError,
+  EncryptionErrorCode,
+} from '../src/encryption/errors';
+import {
+  decryptWithSharedSecret,
+  encryptWithSharedSecret,
+} from '../src/encryption/shared-secret';
+
+// Mock the noble/hashes/utils randomBytes function
+vi.mock('@noble/hashes/utils', () => {
+  // Counter to ensure each call generates different values
+  let counter = 0;
+  return {
+    randomBytes: (size: number) => {
```

Comment:
For my benefit, could you clarify what this is doing?

Reply:
This is to mock this function to make it give random but deterministic values (simi-random). Which is to make debugging the tests easy when needed. Because for every run of the tests the same simi-random value will be used.

Reply:
> random but deterministic values (simi-random)

Does the test only work for a specific order of randomness - is there some limitation in the test that requires determinism?

My overall concern is about the test not actually using proper randomness, which is what would occur in the wild. Just trying to understand why encryption/decryption testing needs to use deterministic keys/randomness...

Similar comment regarding why the shared secret needs to be deterministic (L36-L39)?

Reply:
Many thanks @derekpierre for the discussion.
I did more search and I am more now in favor of having deterministic values for the randomness of our tests here. And probably if we are doing the cryptographic library we would keep this simi-random but have additional completely random values that change with every run.

However, since I am not very much into cryptography related stuff, I will leave this to you (@derekpierre) and @cygnusv to recommend. 
When I first showed @cygnusv this way of generating the nonce, I think he liked it. But I hope him to provide his input, if it is better for it to be like this, or if not, after he read this discussion.

Thanks all,